gaps/gap-14_node_indices_api/gap-14_node_indices_api.md

```diff
@@ -0,0 +1,262 @@
+
+## Definitions
+
+* `node_id` - either `provider_id` or `requestor_id`
+* `node` - participant of the Golem marketplace identified by `node_id`
+* `indices` - any scalar information about a particular `node` or a group of `nodes`
+* `node indices API` - API that could be queried for `indices`. Examples:
+    *   an API that could be queried for "last time online" index for a given `provider_id`
+    *   an API that could be queried for "total amount of all debit notes issued in the last hour"
+
+## Node indices API vs the "reputation" 
+There are three types of information about the Golem market that influence the decisions of the agents trading on the Golem marketplace:
+
+1. Information available directly on the Golem market (e.g. offers and demands)
+2. Information gathered locally by the agent
+3. External information, gathered & provided by other agents
+
+The topic of this GAP is in no way related to 1./2. and should fully cover the most general approach to 3.
+While "local" reputation is part of 2, 3 fully includes everything related to the "non-local" reputation, extended with:
+
+* the node-based indices that really can't be labelled as "reputation" (e.g. how often it is rented, average price etc)
+* the aggregated indices (e.g. average prices for all nodes with a given set of parameters)
+
+## Examples
+
+`Node indices API` might be providing just any information about nodes on the Golem market, but we can safely restrict our attention only to
+types of information that can really be utilized in a market strategy. Few examples of those:
+
+*   Average duration of the agreement for a given `node_id` - we might be looking for a long agreement and prefer nodes that have a history of long agreements
+*   Total amount sent to a given `provider_id` - requestor might want to work with "experienced" providers
+*   Current/past average prices - provider might want to adjust their offer to the current market situation
+*   Node rating provided by the Golem Factory - we might want to work with "good" nodes and trust the Golem Factory scoring methods
+*   All scores of the benchmark performed by the Golem Factory for a given node, with timestamps - ditto, but we only trust Golem Factory benchmarking method, not aggregation
+*   Average traffic on the Golem Market in a given hour of a day - because we expect to find more cheap providers when there's not much going on
+*   How many times given `node_id` shows up in posts on www.reddit.com/r/CheatedOnGolem/ - that's quite obscure, but still - why not?
+
+## Motivation
+
+### Current and future node indices APIs
+
+There are already few examples of node indices APIs that are in progress or planned:
+
+* [Benchmark-based provider reputation (GAP-10)](https://github.com/golemfactory/golem-architecture/pull/33)
+* [Golem Stats](https://stats.golem.network). This API is currently used e.g. by the [community-created provider agent](https://gist.github.com/sv3t0sl4v/28f896752edc9e20347ffc6d8cefe74c)
```

Comment:
i think golem stats is a bad example of this GAP because it uses metric data private to golem factory, making is heavily centralised.